source/slang/slang-parser.cpp

```diff
@@ -1750,6 +1758,51 @@ namespace Slang
             case TokenType::LParent:
                 break;
 
+            case TokenType::OpLess:
+                {
+                    if (parser->options.enableEffectAnnotations)
+                    {
+                        // If we are in a context where effect annotations are allowed,
+                        // then we need to disambiguate the content after "<" to see if it
+                        // should be parsed as an annotation or generic argument list.
+                        // If we can determine the content inside a `<>` is an annotation,
+                        // we will skip the tokens inside the angle brackets.
+                        //
+                        if (parser->tokenReader.peekTokenType() == TokenType::OpLess)
+                        {
+                            if (parser->LookAheadToken("let", 1))
+                            {
+                                // If we see `<let` then we are looking at a generic arg list.
+                            }
+                            else if (parser->LookAheadToken(":", 2))
+                            {
+                                // If we see a "< xxx :", we can also parse it as a generic arg list.
+                            }
+                            else
+                            {
+                                // Otherwise, we may be looking at an effect annotation.
+                                // For now we just skip tokens until we see `>`, if we see any `"` in between,
```

Comment:
This mentions looking for `"`, but the code below looks like it's searching for `;`

Reply:
Yeah, we intend to look for `;` here.